packages/transaction-controller/src/TransactionController.ts

```diff
@@ -776,23 +776,48 @@ export class TransactionController extends BaseController<
             value: '0x0',
           };
 
-    const unsignedEthTx = this.prepareUnsignedEthTx(txParams);
+    const unsignedEthTx = this.prepareUnsignedEthTx(cancelTxParams);
 
     const signedTx = await this.sign(
       unsignedEthTx,
       transactionMeta.txParams.from,
     );
     await this.updateTransactionMetaRSV(transactionMeta, signedTx);
     const rawTx = bufferToHex(signedTx.serialize());
-    await query(this.ethQuery, 'sendRawTransaction', [rawTx]);
-    transactionMeta.estimatedBaseFee = estimatedBaseFee;
-    transactionMeta.status = TransactionStatus.cancelled;
+    const hash = await query(this.ethQuery, 'sendRawTransaction', [rawTx]);
+    const cancelTransactionMeta: TransactionMeta = {
+      ...transactionMeta,
+      actionId,
+      estimatedBaseFee,
+      hash,
+      id: random(),
+      originalGasEstimate: transactionMeta.txParams.gas,
+      status: TransactionStatus.submitted,
+      time: Date.now(),
+      type: TransactionType.cancel,
+      txParams: {
+        ...transactionMeta.txParams,
```

Comment:
As we didn't persist this before, do we need the old params here or are we safe to assume the new ones are complete?